src/combineReducers.js

```diff
@@ -74,6 +74,46 @@ function assertReducerSanity(reducers) {
   })
 }
 
+function getFinalReducers(reducers) {
+  var functionReducers = {}
+
+  Object.keys(reducers).map(key => {
+    if (typeof reducers[key] === 'function') {
+      functionReducers[key] = reducers[key]
+    }
+  })
+
+  return functionReducers
+}
+
+function checkStateChange(keys, finalReducers, state, action) {
```

Comment:
This name is misleading. We’re not just checking the state change, we’re calculating the next state. This is the important part. The fact that we check for changes is just an optimization.